src/chapters/files/csv-files.rst

```diff
@@ -0,0 +1,277 @@
+We pass our ``file`` and the letter ``r`` to the ``open`` function.  More on ``r`` soon.
+
+..TODO:  using with has a special name -- note it here
+
+.. admonition:: Tip
+   
+   ``with`` is a keyword often used with files.  
+   It closes the files once we come to thethe end of the code block.
+
+   When your code block is closed, you can no longer read or write to it.
+
+Reading the Data
+^^^^^^^^^^^^^^^^
+
+Now that you opened your file, let's read it by printing it to console.
+
+.. admonition:: Example
+
+   .. replit:: python
+      :slug: Read-csv-Practice
+      :linenos:
+         
+      import csv
+
+      file = "Working-With-Files/pet-data-Sheet1.csv"
+
+      with open(file, "r") as read_obj:
+         csv_reader = csv.reader(read_obj)
+         
+         for row in csv_reader:
+            print(row)
+
+   **Console Output**
+
+   ::
+
+      ['Name', 'Species', 'Age', 'Size', 'Adoption Date']
+      ['Willow', 'Dog', '6', 'Large', '8/4/2014']
+      ['Alyce', 'Cat', '17', 'Small', '12/19/2014']
+      ['Thisbe', 'Cat', '12', 'Large', '6/12/2009']
+      ['Penny', 'Dog', '4', 'Medium', '8/23/2020']
+      ['Jessie', 'Dog', '12', 'Medium', '12/4/2019']
+      ['Tiny Bubbles', 'Dog', '2', 'Small', '7/1/2021']
+      ['Howser', 'Dog', '5', 'Small', '2/19/2019']
+      ['Baby', 'Dog', '3', 'Extra Large', '3/8/2020']  
+
+
+In line 6, we create another variable to make our code easier to read.  
+We call the ``csv.reader()`` function on our ``read_obj``.
+The ``csv.reader()`` function returns an object that we can iterate through line by line.
+By creating a ``for`` loop in line 8, we to print each row to the console.
+Each row returned by this function is a list of strings.
+
+
+If you only wanted to print data from certain rows, you could modify your ``for`` loop as such:
+
+.. admonition:: Example
+
+   .. sourcecode:: python
+      :lineno-start: 8
+
+      for row in csv.reader(read_obj):
+         print(row[2], row[3])
+
+   **Console Output**
+
+   ::
+
+      Age Size
+      6 Large
+      17 Small
+      12 Large
+      4 Medium
+      12 Medium
+      2 Small
+      5 Small
+      3 Extra Large
+
+
+File Modes
+----------
+
+When we opened this file, we opened it like this:
+
+.. sourcecode:: python
+   :lineno-start: 5
+
+   with open(file, "r") as read_obj:
+
+
+Within the ``open`` function, we pass ``file`` and a string ``r``.
+The ``r`` opens the file in **read mode**.  
+In read mode, we can only work with data already contained within the file.
+
+Other modes include: ``a`` for **append**, ``w`` for **write**, and ``r+`` for **reading and writing**.
+
+Append can be used to add new lines to an existing file or to a new file.
+
+When we write to a file, you need to take care with your files.  
+When using ``w`` you will override whatever file you are writing on.
+This could cause you to lose your original data unless you create a new file to write to.
+
+Writing Data
```

Comment:
both of the csv files in this replit are blank. i think one needs to have data

Reply:
fixed this.